core/src/test/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlanTestSuite.scala

```diff
@@ -120,6 +128,174 @@ class LogicalPlanTestSuite extends BasePlanTest {
+    valueExpectations += List[String]("9223372036854775807", "9223372036854775808")
+    expressionExpectations += LogicalBinaryExpression.and(
+      greaterEqual(new ColumnRef("a", dataType), Constant.create(Long.MaxValue, dataType)),
+      lessEqual(
+        new ColumnRef("a", dataType),
+        Constant.create(new java.math.BigDecimal("9223372036854775808"), dataType)))
+
+    situations += "SELECT a FROM t1 WHERE a<=9223372036854775808 order by a"
+    rowKeyExpectations += List[String](
+      KeyUtils.formatBytesUTF8(RowKey.toRowKey(tableId, new IntHandle(0)).getBytes),
+      KeyUtils.formatBytesUTF8(
+        RowKey.toRowKey(tableId, new IntHandle(Long.MaxValue)).nextPrefix().getBytes),
+      KeyUtils.formatBytesUTF8(
+        RowKey.toRowKey(tableId, new IntHandle(Long.MaxValue + 1)).getBytes),
+      KeyUtils.formatBytesUTF8(
+        RowKey.toRowKey(tableId, new IntHandle(Long.MaxValue + 1)).nextPrefix().getBytes))
+    valueExpectations += List[String]("0", "9223372036854775807", "9223372036854775808")
+    expressionExpectations += lessEqual(
+      new ColumnRef("a", dataType),
+      Constant.create(new java.math.BigDecimal("9223372036854775808"), dataType))
+
+    situations += "SELECT a FROM t1 WHERE a <= 9223372036854775807 order by a"
+    rowKeyExpectations += List[String](
+      KeyUtils.formatBytesUTF8(RowKey.toRowKey(tableId, new IntHandle(0)).getBytes),
+      KeyUtils.formatBytesUTF8(
+        RowKey.toRowKey(tableId, new IntHandle(Long.MaxValue)).nextPrefix().getBytes))
+    valueExpectations += List[String]("0", "9223372036854775807")
+    expressionExpectations += lessEqual(
+      new ColumnRef("a", dataType),
+      Constant.create(Long.MaxValue, dataType))
+
+    situations += "SELECT a FROM t1 order by a"
+    rowKeyExpectations += List[String](
+      KeyUtils.formatBytesUTF8(RowKey.createMin(tableId).getBytes),
+      KeyUtils.formatBytesUTF8(RowKey.createBeyondMax(tableId).getBytes))
+    valueExpectations += List[String](
+      "0",
+      "9223372036854775807",
+      "9223372036854775808",
+      "18446744073709551615")
+
+    for (i <- situations.indices) {
+      val df = spark.sql(situations(i))
+      val dag = extractDAGRequests(df).head
+      val rangeFilter = dag.getRangeFilter
+      val range = dag.getRangesMaps.get(tableId)
+      val value = df.collect()
+      for (x <- 0 until range.size()) {
+        val startKey = KeyUtils.formatBytesUTF8(range.get(x).getStart.toByteArray)
+        val endKey = KeyUtils.formatBytesUTF8(range.get(x).getEnd.toByteArray)
+        assert(startKey.equals(rowKeyExpectations(i)(2 * x)))
+        assert(endKey.equals(rowKeyExpectations(i)(2 * x + 1)))
+      }
+      for (x <- value.indices) {
+        assert(value(x)(0).toString.equals(valueExpectations(i)(x)))
+      }
+      if (rangeFilter.size() == 1) {
+        println(i)
+        assert(rangeFilter.get(0).toString.equals(expressionExpectations(i).toString))
+      }
+    }
+  }
+
```

Comment:
add test:
- `SELECT a FROM t1 WHERE a >= 9223372036854775808 and a<=9223372036854775809`
- `SELECT a FROM t1 WHERE a >= 9223372036854775808`

Reply:
> add test:
> 
> * `SELECT a FROM t1 WHERE a >= 9223372036854775808 and a<=9223372036854775809`
> * `SELECT a FROM t1 WHERE a >= 9223372036854775808`

add test